api/tests/Feature/CandidateAssessmentStatusTest.php

```diff
@@ -504,4 +505,124 @@ public function testOutOfOrderAssessmentsDoNotIncrementStep()
                 ],
             ]);
     }
+
+    public function testMultipleEssentialMustPassToIncrementStep()
+    {
+
+        $pool = Pool::factory()
+            ->published()
+            ->create([
+                'team_id' => $this->team->id,
+            ]);
+
+        $technicalSkills = Skill::where('category', SkillCategory::TECHNICAL->name)->limit(2)->get();
+        $poolSkillOne = PoolSkill::create([
+            'pool_id' => $pool->id,
+            'skill_id' => $technicalSkills[0]->id,
+            'type' => PoolSkillType::ESSENTIAL->name,
+        ]);
+
+        // Non-essential skills must also be assessed if a user claims them
+        $poolSkillTwo = PoolSkill::create([
+            'pool_id' => $pool->id,
+            'skill_id' => $technicalSkills[1]->id,
+            'type' => PoolSkillType::NONESSENTIAL->name,
+        ]);
+
+        $user = User::factory()->create();
+        UserSkill::factory()->create([
```

Comment:
Do you not need to give the user the first skill as well?

Reply:
We don't need to. The logic only cares that the skill was assessed, it doesn't really care if the user claimed it or not. We assume the user has claimed the necessary skills since it is required to submit an application.

Reply:
Right!